src/Utils/Color.vala

```diff
@@ -69,4 +72,38 @@ public class Akira.Utils.Color : Object {
 
         return uint_rgba;
     }
+
+    public static Gdk.RGBA blend_colors (Gdk.RGBA base_color, Gdk.RGBA added_color) {
+        // If the newly added color alpha is 0 we don't need to do any color mixing
+        // as the added color won't alter the base color.
+        if (added_color.alpha == 0.0) {
+            return base_color;
+        }
+
+        // If the newly added color alpha is 1 or the base color alpha is 0
+        // we don't need to do any color mixing, as the added color will be
+        // the only one visible.
+        if (added_color.alpha == 1.0 || base_color.alpha == 0) {
+            return added_color;
+        }
+
+        // Simple blend using Alpha channels. In the future we will support
+        // different blending modes.
+        double alpha = 1 - (1 - added_color.alpha) * (1 - base_color.alpha);
+
+        double red = added_color.red * added_color.alpha /
```

Comment:
You can stroe the alpha rations for performance.

double ar1 = added_color.alpha / alpha;
doouble ar2 = (1 - added_color.alpha) / alpha

Reply:
Ah, good catch.